### PR TITLE
CSCQVAIN-163: Added license header placeholders.

### PR DESCRIPTION
**Pull request by jppiiroinen**
Added license headers placeholders for all files.

Added 'make headers' which will scan thru the files and
updates all the files with up to date information.

Supported placeholder formats are:
```
/* ADD_LICENSE_HEADER */
<!-- ADD_LICENSE_HEADER -->
# ADD_LICENSE_HEADER
```

The license header itself comes from .license-header.

### CODE REVIEW

**Comment by jppiiroinen** [comment]
```
diff --git a/src/App.vue b/src/App.vue
index fc2ddde..fc3a35a 100644
--- a/src/App.vue
+++ b/src/App.vue
@@ -1,4 +1,16 @@
-<!-- ADD_LICENSE_HEADER -->
+<!--
+This file is part of Qvain -project.
+
+Author(s):
+       Wouter Van Hemel <wouter.van.hemel@helsinki.fi>
+       Jori Niemi <3295718+tahme@users.noreply.github.com>
+
+License: GPLv3
+
+See LICENSE file for more information.
+Copyright (C) 2019 Ministry of Culture and Education, Finland.
+All Rights Reserved.
+-->
 <template>
```

**Comment by jppiiroinen** [comment]
with this release flow there would be a release branch which would have those in place:
```
------o--------------------------------master
     /               
----o--------------------------------- release
   /   <- updated headers, added tag
--o---------------------------------- next
```
this would mean that while we work (next branch) on the code we just have those placeholders and only on release we set them as we are not going to change those files anymore (release, master).

**Comment by jppiiroinen** [comment]
```
$ time make headers
[OK] ./scripts/deref.js
[OK] ./scripts/access_type.js
[OK] ./scripts/funder_type.js
[OK] ./tmp/obj/obj.js
[OK] ./tmp/json-schema-live/build/keywords.js
[OK] ./tmp/json-schema-live/build/is.js
[OK] ./tmp/json-schema-live/build/test.js
[OK] ./tmp/json-schema-live/build/validity.js
[OK] ./tmp/json-schema-live/build/index.js
[OK] ./tmp/json-schema-live/build/combiners.js
[OK] ./tmp/json-schema-live/build/config.js
[OK] ./tmp/json-schema-live/build/error.js
[OK] ./tmp/json-schema-live/build/data_validators.js
[OK] ./tmp/json-schema-live/build/pointer.js
[OK] ./tmp/json-schema-live/build/count.js
[OK] ./tmp/json-schema-live/build/testschemas.js
[OK] ./tmp/json-schema-live/build/deep_equal.js
[OK] ./tmp/json-schema-live/build/type_validators.js
[OK] ./tmp/json-schema-live/build/walk.js
[OK] ./tmp/json-schema-live/build/validate.js
[OK] ./tmp/json-schema-live/build/live.js
[OK] ./tmp/json-schema-live/src/keywords.js
[OK] ./tmp/json-schema-live/src/is.js
[OK] ./tmp/json-schema-live/src/test.js
[OK] ./tmp/json-schema-live/src/validity.js
[OK] ./tmp/json-schema-live/src/index.js
[OK] ./tmp/json-schema-live/src/combiners.js
[OK] ./tmp/json-schema-live/src/config.js
[OK] ./tmp/json-schema-live/src/error.js
[OK] ./tmp/json-schema-live/src/data_validators.js
[OK] ./tmp/json-schema-live/src/pointer.js
[OK] ./tmp/json-schema-live/src/count.js
[OK] ./tmp/json-schema-live/src/testschemas.js
[OK] ./tmp/json-schema-live/src/deep_equal.js
[OK] ./tmp/json-schema-live/src/type_validators.js
[OK] ./tmp/json-schema-live/src/walk.js
[OK] ./tmp/json-schema-live/src/validate.js
[OK] ./tmp/json-schema-live/src/live.js
[OK] ./tmp/tmp.js
[OK] ./tmp/schematodata.js
[OK] ./tmp/flatten.js
[OK] ./tmp/listschemas.js
[OK] ./tmp/datapointer.js
[OK] ./tmp/jptr.js
[OK] ./tmp/validate_backup.js
[OK] ./tmp/deep.js
[OK] ./tmp/validate.js
[OK] ./vendor/uris/uris-debug.js
[OK] ./vendor/uris/uris.js
[OK] ./vendor/validator/test/test.js
[OK] ./vendor/validator/src/keywords.js
[OK] ./vendor/validator/src/is.js
[OK] ./vendor/validator/src/validity.js
[OK] ./vendor/validator/src/combiners.js
[OK] ./vendor/validator/src/deepcopy.js
[OK] ./vendor/validator/src/error.js
[OK] ./vendor/validator/src/data_validators.js
[OK] ./vendor/validator/src/pointer.js
[OK] ./vendor/validator/src/count.js
[OK] ./vendor/validator/src/testschemas.js
[OK] ./vendor/validator/src/type_validators.js
[OK] ./vendor/validator/src/walk.js
[OK] ./vendor/validator/src/validate.js
[OK] ./vendor/murmurhash/murmurhash3.js
[OK] ./vendor/murmurhash/murmurhash3-node.js
[OK] ./vendor/json-pointer/test/test.js
[OK] ./vendor/json-pointer/index.js
[FIX] ./src/mixins/border-color-mixin.js
[FIX] ./src/auth/token.vue
[FIX] ./src/auth/auth.js
[FIX] ./src/auth/old.js
[FIX] ./src/auth/jwt.js
[FIX] ./src/auth/plugin.js
[FIX] ./src/auth/busy.vue
[FIX] ./src/orcid/search.vue
[FIX] ./src/orcid/api.js
[FIX] ./src/person/person.vue
[FIX] ./src/person/list.vue
[FIX] ./src/composites/RecordField.vue
[FIX] ./src/utils/debounce.js
[FIX] ./src/schemas/bundle.js
[FIX] ./src/schemas/fairdata.bundle.js
[FIX] ./src/schemas/fairdata-att.ui.diff.js
[FIX] ./src/schemas/fairdata-ida.ui.js
[FIX] ./src/components/PreservationState.vue
[FIX] ./src/components/PreservationStateBadge.vue
[FIX] ./src/components/LanguageSelect.vue
[FIX] ./src/components/Loader.vue
[FIX] ./src/components/i18nStrings.vue
[FIX] ./src/components/BusyButton.vue
[FIX] ./src/components/DateRange.vue
[FIX] ./src/components/TagList.vue
[FIX] ./src/components/ReferenceData.vue
[FIX] ./src/components/i18nTextarea.vue
[FIX] ./src/components/Overview.vue
[FIX] ./src/components/DatasetJsonModal.vue
[FIX] ./src/components/DatasetOverviewModal.vue
[FIX] ./src/components/StringList.vue
[FIX] ./src/components/PublishModal.vue
[FIX] ./src/components/Date.vue
[FIX] ./src/components/VersionsModal.vue
[FIX] ./src/components/ValidationPopover.vue
[FIX] ./src/components/EmptyNote.vue
[FIX] ./src/components/Wrapper.vue
[FIX] ./src/vuex/files.js
[FIX] ./src/vuex/editor.js
[FIX] ./src/viewer/HintEditor.vue
[FIX] ./src/viewer/SchemaTree.vue
[FIX] ./src/viewer/v-schema-viewer.vue
[FIX] ./src/filebrowser/PasMetadata.vue
[FIX] ./src/filebrowser/fileinfo-modal.vue
[FIX] ./src/filebrowser/FileItem.vue
[FIX] ./src/filebrowser/file.vue
[FIX] ./src/filebrowser/breadcrumbs.vue
[FIX] ./src/filebrowser/tree.vue
[FIX] ./src/filebrowser/FileEditModal.vue
[FIX] ./src/filebrowser/FilePicker.vue
[FIX] ./src/filebrowser/Browser.vue
[FIX] ./src/lib/foreach_schema.mjs
[FIX] ./src/lib/pruneEmpty.js
[FIX] ./src/lib/test.mjs
[FIX] ./src/lib/keysWithOrder.js
[FIX] ./src/lib/deepCopy.js
[FIX] ./src/lib/genid.js
[FIX] ./src/lib/cloneWithPrune.js
[FIX] ./src/lib/foreach_schema.js
[FIX] ./src/api/client.js
[FIX] ./src/api/api.js
[FIX] ./src/exp/tags.vue
[FIX] ./src/views/TabUi.vue
[FIX] ./src/views/Config.vue
[FIX] ./src/views/Editor.vue
[FIX] ./src/views/UserInfo.vue
[FIX] ./src/views/Datasets.vue
[FIX] ./src/views/SingleTab.vue
[FIX] ./src/views/Welcome.vue
[FIX] ./src/views/RecordList.vue
[FIX] ./src/partials/DeleteButton.vue
[FIX] ./src/partials/Title.vue
[FIX] ./src/partials/ValidationFailedIcon.vue
[FIX] ./src/partials/InfoIcon.vue
[FIX] ./src/partials/ValidationUnclearIcon.vue
[FIX] ./src/partials/ValidationSuccessIcon.vue
[FIX] ./src/partials/ValidationStatus.vue
[FIX] ./src/widgets/refdata/list.vue
[FIX] ./src/widgets/refdata/list-ui.vue
[FIX] ./src/widgets/refdata/es.js
[FIX] ./src/widgets/refdata/autocomplete.vue
[FIX] ./src/widgets/extra/widget-googlemaps.vue
[FIX] ./src/widgets/Number.vue
[FIX] ./src/widgets/FlatObject.vue
[FIX] ./src/widgets/editable-list.vue
[FIX] ./src/widgets/TabbedArray.vue
[FIX] ./src/widgets/allOf.vue
[FIX] ./src/widgets/skip.js
[FIX] ./src/widgets/Enum.vue
[FIX] ./src/widgets/InlineArray.vue
[FIX] ./src/widgets/oneOf.vue
[FIX] ./src/widgets/Array.vue
[FIX] ./src/widgets/anyOf.vue
[FIX] ./src/widgets/mapping.js
[FIX] ./src/widgets/base.vue
[FIX] ./src/widgets/plugin-ui-refdata.js
[FIX] ./src/widgets/SelfReferentialObject.vue
[FIX] ./src/widgets/widget-googlemaps.vue
[FIX] ./src/widgets/String.vue
[FIX] ./src/widgets/Object.vue
[FIX] ./src/widgets/TabSelector.vue
[FIX] ./src/widgets/plugin-ui-extra.js
[FIX] ./src/App.vue
[FIX] ./src/store.js
[FIX] ./src/schema_to_tabs.js
[FIX] ./src/main.js
[FIX] ./src/testschemas_ui.js
[FIX] ./src/router.js
[FIX] ./src/walkjson.js
[FIX] ./src/testschemas_data.js
[FIX] ./src/testschemas.js
[FIX] ./src/navigation.vue
[FIX] ./babel.config.js
[OK] ./update-header.py
[FIX] ./vue.config.js
[FIX] ./webpack-old.config.js

real	0m15.917s
user	0m4.591s
sys	0m4.100s
```

**Review by tahme — APPROVED** [review]

File: update-header.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                        f.seek(0)
+                        f.write(data)
+
+                if was_found:
+                    print("[FIX] {filename}".format(filename=file_name))
+                else:
+                    print("[OK] {filename}".format(filename=file_name))
+
+
+    #with open(sys.argv[1])
```
Comment (jppiiroinen) [inline update-header.py]: a bug

Comment (jppiiroinen) [reply update-header.py]: fixed

**Pull request merged**